File: app/src/main/java/com/gee12/mytetroid/views/dialogs/IntentDialog.java
package com.gee12.mytetroid.views.dialogs;

import android.content.Context;
import android.widget.ListView;

import androidx.appcompat.app.AlertDialog;

import com.gee12.htmlwysiwygeditor.Dialogs;
import com.gee12.mytetroid.R;
import com.gee12.mytetroid.views.adapters.IntentsAdapter;
import com.gee12.mytetroid.model.ReceivedData;

@Deprecated
public class IntentDialog {

    public interface IItemClickListener {
        void onItemClick(ReceivedData item);
    }

    /**
     * TODO: Перенести в DialogFragment.
     * Диалог со списком вариантов обработки переданного объекта.
     * @param context
     * @param isText
     * @param listener
     */
    public static void createDialog(Context context, boolean isText, IItemClickListener listener) {
        Dialogs.AskDialogBuilder builder = Dialogs.AskDialogBuilder.create(context, R.layout.dialog_list_view);
//        builder.setTitle("Выберите действие");
        final AlertDialog dialog = builder.create();

        ListView listView = builder.getView().findViewById(R.id.list_view);
        ReceivedData[] dataSet = (isText) ? ReceivedData.textIntents() : ReceivedData.imageIntents();
        listView.setOnItemClickListener((parent, view, position, id) -> {
            listener.onItemClick(dataSet[position]);
            dialog.cancel();
        });
        listView.setAdapter(new IntentsAdapter(context, dataSet));
        dialog.show();
    }
}

File: app/src/main/java/com/gee12/mytetroid/views/dialogs/IntentDialog.kt
package com.gee12.mytetroid.views.dialogs

import android.view.View
import android.widget.AdapterView
import android.widget.ListView
import androidx.appcompat.app.AlertDialog
import com.gee12.mytetroid.R
import com.gee12.mytetroid.model.ReceivedData
import com.gee12.mytetroid.viewmodels.StorageViewModel
import com.gee12.mytetroid.views.adapters.IntentsAdapter
import org.koin.java.KoinJavaComponent.get

/**
 * Диалог со списком вариантов обработки переданного объекта.
 */
class IntentDialog(
    private val isText: Boolean,
    private val onItemClick: (item: ReceivedData) -> Unit,
) : TetroidDialogFragment<StorageViewModel>() {

    override fun getRequiredTag() = TAG

    override fun isPossibleToShow() = true

    override fun getLayoutResourceId() = R.layout.dialog_list_view

    override fun createViewModel() {
        this.viewModel = get(StorageViewModel::class.java)
    }

    override fun onDialogCreated(dialog: AlertDialog, view: View) {
//        builder.setTitle("Выберите действие");
        val listView = view.findViewById<ListView>(R.id.list_view)
        val dataSet = if (isText) ReceivedData.textIntents() else ReceivedData.imageIntents()
        listView.onItemClickListener = AdapterView.OnItemClickListener { _: AdapterView<*>?, _: View?, position: Int, _: Long ->
            onItemClick(dataSet[position])
            dialog.cancel()
        }
        listView.adapter = IntentsAdapter(context, dataSet)
    }

    companion object {
        const val TAG = "IntentDialog"
    }
}